server/wal/treemap.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wal
+
+import (
+	"fmt"
+
+	rbt "github.com/emirpasic/gods/trees/redblacktree"
+	"github.com/emirpasic/gods/utils"
+)
+
+type treeMap[K comparable, V any] struct {
```

Comment:
Would it make sense to just switch to a map implementation that already supports generics? eg: https://github.com/zyedidia/generic/tree/master/btree

Reply:
`gods` seems have 10x star than `zyedidia/generic`, and if we switch to `zyedidia/generic`, we should implement min、max、floor by our self.

I have searched for an generic data structure implementation in golang, and didn't found any powerful as `gods` library.

Reply:
Sounds good. Thanks for checking!